pygfx/renderers/wgpu/shader/base.py

```diff
@@ -233,16 +239,22 @@ def define_colormap(self, texture, texcoords, interpolation="linear"):
             Binding("t_colormap", "texture/auto", texture_view, "FRAGMENT"),
         ]
 
-    def define_img_colormap(self, texture, interpolation="linear"):
+    def define_img_colormap(self, map):
         """Define the given texture view as the colormap to be used to
         lookup the final color from the image data.
         In the WGSL the colormap can be sampled using ``sample_colormap()``.
         Returns a list of bindings.
         """
 
-        self["colormap_interpolation"] = interpolation
-        sampler = GfxSampler(interpolation, "clamp")
-        texture_view = self._define_colormap_texture(texture)
+        self["colormap_interpolation"] = (
+            map.mag_filter
+        )  # todo: remove colormap_interpolation?
```

Comment:
Currently, the colormap has some special sampler logic compared to other texturemaps. If I understand correctly, this is to enable the colormap to support different data formats? 
Is there any specific consideration for using `textureLoad` instead of `textureSample` for sampling certain colormap textures in WGSL?

Reply:
> The use of `textureLoad` is to sample from textures that have a format for which we cannot create a sampler. This is determined by `colormap_format` being `f32 / u32 / i32`.

`textureSample` can also be used with `u32` and `i32` texture formats as long as `GPUTextureSampleType` is appropriately set to `uint` or `sint`. However, for integer-format textures, interpolation and filtering are not supported, which aligns with the behavior of `textureLoad`. That said, this method provides a unified API with floating-point textures, making the code simpler and more convenient. 

Additionally, if interpolation and filtering operations are required for integer-format textures, it is recommended to preprocess the texture by converting it into a floating-point format, as filtered sample values are inherently floating-point.

Reply:
If you agree, I plan to attempt a refactor of the colormap-related logic in a separate PR, with the following goals:

- Unify the code logic for all `TextureMap` types, so that `colormap` behaves consistently with all other maps and no longer requires special handling.

- Add support for setting `alpha_mode` ("opaque", "mask", "blend") and `alpha_test` (`alpha_cut_off`) to allow us to pass the test cases available [here](https://github.com/KhronosGroup/glTF-Sample-Assets/blob/main/Models/AlphaBlendModeTest/README.md).

- Clean up the code related to Vertex Color mode to ensure compatibility with the test cases provided [here](https://github.com/KhronosGroup/glTF-Sample-Assets/blob/main/Models/VertexColorTest/README.md).

Reply:
Sounds good!